Song/VerseWriter.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Text;
+
+namespace Song
+{
+    public class VerseWriter
+    {
+        private string _firstLine;
+
+        private string _uniqueLine;
+
+        private List<string> _middleLines;
+
+        private string _lastLine;
+
+        private string _lastAnimalName;
+
+        public VerseWriter() 
+        {
+            _firstLine = string.Empty;
+            _uniqueLine = string.Empty;
+            _middleLines = new List<string>();
+            _lastLine = string.Empty;
+            _lastAnimalName = string.Empty;
+        }
+        public virtual string WriteVerse(string animalName, string animalUniqueLine)
```

Comment:
This should be further split to another component - which prints first, last and rhyme lines. This way, each of those methods can be tested individually and the tests of this class will be trivial and you will need less of them. Right now, I don;t like the tests, because they are very repetitve and don't give away the patterns that easily.